tutorials/Tutorial17_GPL.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+"""Generative-Pseudo-Labeling-for-Domain-Adaptation-of-Dense-Retrievals.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/drive/1Tz9GSzre7JfvXDDKe7sCnO0FMuDViMnN
+
+# Generative Pseudo Labeling for Domain Adaptation of Dense Retrievals
+#### Note: Adapted to Haystack from Nils Riemers' original [notebook](https://colab.research.google.com/gist/jamescalam/d2c888775c87f9882bb7c379a96adbc8/gpl-domain-adaptation.ipynb#scrollTo=183ff7ab)
+
+NLP models we use every day were trained on a large corpus of data, reflecting the World from the past. What if some significant world-changing events occur in the meantime, and we want our models to know about them? Like the COVID pandemic, for example? It doesn't make sense to retrain the models from scratch as that would be wasteful. What if we could update the models with new essential data instead of retraining the models from scratch? **Generative Pseudo Labeling (GPL)** to the rescue.
+
+In the below example, we demonstrate this for a simple query: "How is COVID-19 transmitted?".
+
+As model, we use TAS-B: A DistilBERT model that achieves state-of-the-art performance on MS MARCO (500k queries from Bing Search Engine). Both DistilBERT and MS MARCO were created with data from 2018 and before, hence, it lacks the knowledge of any COVID-related information.
+
+In your example we use a small collection of just 4 documents. If you search with this model, you get the following results (dot-score & document):
+- 94.84	Ebola is transmitted via direct contact with blood
+- 92.87	HIV is transmitted via sex or sharing needles
+- 92.31	Corona is transmitted via the air
+- 91.54	Polio is transmitted via contaminated water or food
+
+
+As we see, the correct document is just ranked on 3rd place behind how Ebola and HIV are transmitted.
```

Comment:
You can see that the correct document is only third, outranked by Ebola and HIV information. Let's see how we can make this better.